pkg/operator/status.go

```diff
@@ -393,6 +393,25 @@ func (c *Controller) syncStatus(
 
 	updateCondition(cr, defaults.OperatorStatusTypeRemoved, operatorRemoved)
 
+	// ensure the operator conditions will not remain progressing or degraded
+	// because the azure path fix job hasn't completed or has failed.
+	// when the operator is set to Removed, it will delete the storage
+	// account that it provisioned (unless .spec.storage.managementState is
+	// Unmanaged). since the storage account is (or will be) removed, there
+	// is no point in reporting the state of the azure path fix job.
+	if cr.Spec.ManagementState == operatorapiv1.Removed && cr.Spec.Storage.ManagementState != imageregistryv1.StorageManagementStateUnmanaged {
```

Comment:
If we expect an update from the job informer to cleanup the conditions, shouldn't we also cleanup the job itself?

Reply:
I figured it would probably get GC'd, but that's sloppy of me.
If I remove the early return on https://github.com/openshift/cluster-image-registry-operator/pull/1142/files/163815b88a48d7396a5b92ccc7da0c0e1153aa8d#diff-6b69f0f2a6978665596dfc3f74d52fd4a1aa430198d1f7fb1b84e6b773c9fbc4R182 then I think that should be taken care of. 
WDYT?

Reply:
It would be if it was created with that in mind: https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/

Reply:
Ah then we must do this manually indeed. I'm on it.

Reply:
I think this might no longer be required after removing the early return on https://github.com/openshift/cluster-image-registry-operator/pull/1142/files#diff-6b69f0f2a6978665596dfc3f74d52fd4a1aa430198d1f7fb1b84e6b773c9fbc4R182 🤔 
When management state is set to Removed, the AzurePathFix controller will run (because it watches the image registry config). Setting management state to Removed clears up `.status.storage`, and since the controller is now allowed to run even though that is no longer set, the job will get removed and the conditions cleared.

I will comment this code for now and run some tests. Things will need to look a bit different in the AzurePathFix controller on 4.14-4.16 (I think we might need to explicitly check for `.spec.managementState: Removed`), but I think that even then we won't need this code block here.

I'll report back with my findings.

Reply:
I adapted the code here and applied it onto 4.16: https://github.com/openshift/cluster-image-registry-operator/pull/1158
My tests showed that the fix works without changes to `status.go`. The only changes required are the removal of the early return in cases when storage status is unset, and specific handling of `managementState: Removed`.
I have cleaned up the commits in this PR to reflect these findings.
PTAL @atiratree